osi_ippay_payment/models/payment_transaction.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import requests
+import xmltodict
+import json
+import logging
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTansaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    @api.multi
+    def _ippay_s2s_do_payment(self, invoice):
+        amount = str(self.amount).replace('.', '')
+        token_data = invoice.payment_token_id.acquirer_ref
+        request = '''
+            <ippay>
+                <TransactionType>SALE</TransactionType>
+                <TerminalID>TESTTERMINAL</TerminalID>
+                <Token>%s</Token>
+                <TotalAmount>%s</TotalAmount>
+            </ippay>''' % (token_data, int(amount))
+
+        _logger.info("Request to get IPPay Transaction ID: %s" % (request))
+        url = 'https://testgtwy.ippay.com/ippay'
```

Comment:
The URL should depend on IPPay's configuration - test or production.

Reply:
I agree, this should not be hardcoded but perhaps tied to a setting for test/production.